### PR TITLE
KEYCLOAK-16063 Messed output when running './kc.sh'

### PR DESCRIPTION
**Pull request by mposolda**

### CODE REVIEW

**Comment by mposolda** [comment]
@pedroigor With this change, the output of the "./kc.sh" looks like this:

```
$ ./kc.sh 
2020-10-23 18:42:57,126 ERROR [org.key.cli.Picocli] (main) ERROR: Failed to start server using profile (none).
2020-10-23 18:42:57,127 ERROR [org.key.cli.Picocli] (main) ERROR: Key material not provided to setup HTTPS. Please configure your keys/certificates or enable HTTP or start the server using the 'dev' profile.
2020-10-23 18:42:57,127 ERROR [org.key.cli.Picocli] (main) For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.
```

The output with more errors:
```
$./kc.sh --proxy=foo
2020-10-23 18:44:10,008 ERROR [org.key.cli.Picocli] (main) ERROR: Failed to start server using profile (none).
2020-10-23 18:44:10,008 ERROR [org.key.cli.Picocli] (main) ERROR: Multiple configuration errors during startup
2020-10-23 18:44:10,009 ERROR [org.key.cli.Picocli] (main) ERROR 1
2020-10-23 18:44:10,009 ERROR [org.key.cli.Picocli] (main) ERROR: Key material not provided to setup HTTPS. Please configure your keys/certificates or enable HTTP or start the server using the 'dev' profile.
2020-10-23 18:44:10,009 ERROR [org.key.cli.Picocli] (main) ERROR 2
2020-10-23 18:44:10,009 ERROR [org.key.cli.Picocli] (main) ERROR: Invalid value [foo] for configuration property [proxy].
2020-10-23 18:44:10,009 ERROR [org.key.cli.Picocli] (main) For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.
```

File: quarkus/runtime/src/main/java/org/keycloak/provider/quarkus/QuarkusConfigurationException.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+package org.keycloak.provider.quarkus;
+
+/**
+ * Exception thrown when some error happens during initialization of Quarkus platform. Usually due the incorrect configuration of basic stuff (DB, HTTP protocol etc)
+ * This exception is supposed to be thrown when Quarkus platform is started (See {@link io.quarkus.runtime.StartupEvent}
+ *
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class QuarkusConfigurationException extends RuntimeException {
```
Comment (pedroigor) [inline quarkus/runtime/src/main/java/org/keycloak/provider/quarkus/QuarkusConfigurationException.java:27]: Maybe `InitializationException` ?

File: quarkus/runtime/src/main/java/org/keycloak/configuration/Messages.java [hunk 2]
```diff
@@ -30,6 +30,11 @@ static IllegalArgumentException invalidProxyMode(String mode) {
     }
 
     static IllegalStateException httpsConfigurationNotSet() {
-        return new IllegalStateException("Key material not provided to setup HTTPS. Please configure your keys/certificates or enable HTTP.");
+        StringBuilder builder = new StringBuilder("Key material not provided to setup HTTPS. Please configure your keys/certificates or enable HTTP");
+        if (!"dev".equals(Environment.getProfile())) {
```
Comment (pedroigor) [inline quarkus/runtime/src/main/java/org/keycloak/configuration/Messages.java:34]: Can we create an `Environment#isDevMode` ?

**Review by pedroigor — COMMENTED** [review]
@mposolda That is great stuff ... I really liked how exceptions are being handled now ... Much better.

Just a few comments and nitpicks.

**Review by pedroigor — APPROVED** [review]
Let's do any of these changes in another round, if necessary.

**Pull request merged**